event-publisher/src/publisher.rs

```diff
@@ -0,0 +1,36 @@
+use mbus_api::{event::EventMessage, mbus_nats::message_bus_init, Bus};
+
+/// Initialize the mbus. Connects to the mbus server and creates a stream if not exists.
+pub(crate) async fn mbus_init(mbus_url: &str) -> impl Bus {
+    message_bus_init(mbus_url).await
+}
+
+/// Message bus publisher.
+#[derive(Clone)]
+pub(crate) struct MbusPublisher {}
+
+impl MbusPublisher {
+    /// Initializes mbus and starts the publisher.
+    pub async fn start(mbus_url: &str, mut recv: tokio::sync::mpsc::Receiver<EventMessage>) {
```

Comment:
start usually means it starts on another thread and exists but here seems like it's not, so perhaps run is more appropriate.